src/renderer/tvgCanvas.h

```diff
@@ -104,12 +108,13 @@ struct Canvas::Impl
             }
             refresh = false;
         }
+        status = Status::Updating;
```

Comment:
update() is also called from push(). Therefore, viewport() only operates when the canvas is synchronized(or clean?). after creation and sync(). If this is the intended behavior, I think it would be a good to describe it to the docs.